Document the Kubeflow Pipelines SDK for Tekton (#2341)
kubeflow/website: content/en/docs/pipelines/sdk/pipelines-with-tekton.md

+++
title = "Kubeflow Pipelines SDK for Tekton"
description = "How to run Kubeflow Pipelines with Tekton"
weight = 140
                    
+++

> By default Kubeflow Pipelines get compiled into [Argo Workflow Templates](https://github.com/argoproj/argo/blob/master/docs/workflow-concepts.md)

Reviewer: Remove argo references from this doc

Author: Done.

  * `kfp_tekton.TektonClient.upload_pipeline` uploads a local file to create a new pipeline in Kubeflow Pipelines.
  * `kfp_tekton.TektonClient.create_experiment` creates a pipeline
    [experiment](/docs/pipelines/concepts/experiment/) and returns an
    experiment object.
  * `kfp_tekton.TektonClient.run_pipeline` runs a pipeline and returns a run object.
  * `kfp_tekton.TektonClient.create_run_from_pipeline_func` compiles a pipeline
    function and submits it for execution on Kubeflow Pipelines.
  * `kfp_tekton.TektonClient.create_run_from_pipeline_package` runs a local 
    pipeline package on Kubeflow Pipelines.

## Installing the KFP-Tekton SDK

You need **Python 3.5** or later to use the Kubeflow Pipelines SDK for Tekton.
We recommend to create a Python virtual environment first using
[Miniconda](https://conda.io/miniconda.html) or a virtual environment
manager such as `virtualenv` or the Python 3 `venv` module:

    python3 -m venv .venv-kfp-tekton
    source .venv-kfp-tekton/bin/activate

You can install the latest release of the `kfp-tekton` compiler from
[PyPi](https://pypi.org/project/kfp-tekton/):
    

>     pip3 install kfp-tekton --upgrade

Reviewer 2: ```suggestion
    ```shell
    pip3 install kfp-tekton --upgrade
    ```
```

(reference: https://kubernetes.io/docs/contribute/style/style-guide/#inline-code-formatting)

> ## Compiling Kubeflow Pipelines DSL Scripts

Reviewer 2: If scripts doesn't have to be capitalized, then:

```suggestion
## Compiling Kubeflow Pipelines DSL scripts
```

(sentence-case)

The `kfp-tekton` Python package comes with the `dsl-compile-tekton` command line
executable, which should be available in your terminal shell environment after
installing the `kfp-tekton` Python package.

If you cloned the `kfp-tekton` project, you can find example pipelines in the

> `samples` folder or under `sdk/python/tests/compiler/testdata` folder.

Reviewer 2: ```suggestion
`samples` folder or in the `sdk/python/tests/compiler/testdata` folder.
```

>     dsl-compile-tekton \
>         --py sdk/python/tests/compiler/testdata/parallel_join.py \
>         --output pipeline.yaml

Reviewer 2: Check if you can wrap YAML in a code block.

For example:

```suggestion
    ```yaml
    dsl-compile-tekton \
        --py sdk/python/tests/compiler/testdata/parallel_join.py \
        --output pipeline.yaml
    ```
```

if the `yaml` bit before three backticks doesn't work, maybe it'd stil make sense to use three backticks (reference: https://kubernetes.io/docs/contribute/style/style-guide/#inline-code-formatting)

**Note**: If the KFP DSL script contains a `__main__` method calling the
`kfp_tekton.compiler.TektonCompiler.compile()` function:

```Python
if __name__ == "__main__":
    from kfp_tekton.compiler import TektonCompiler
    TektonCompiler().compile(pipeline_func, "pipeline.yaml")
```

The pipeline can then be compiled by running the DSL script with `python3`
directly from the command line, producing a Tekton YAML file `pipeline.yaml`
in the same directory:

>     python3 pipeline.py

Reviewer 2: ```suggestion
    ```shell
    python3 pipeline.py
    ```
```

https://kubernetes.io/docs/contribute/style/style-guide/#inline-code-formatting

> ## Next steps

Reviewer: ```suggestion
## Additional Documentation
```

Author: Done
